crates/re_log_types/src/path/parse_path.rs

```diff
@@ -150,44 +139,60 @@ impl std::str::FromStr for DataPath {
+    pub fn parse_strict(s: &str) -> Result<Self, PathParseError> {
+        let DataPath {
+            entity_path,
+            instance_key,
+            component_name,
+        } = DataPath::from_str(s)?;
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        if STRICT_ENTITY_PATH_PARSING {
-            let DataPath {
-                entity_path,
-                instance_key,
-                component_name,
-            } = DataPath::from_str(s)?;
-
-            if let Some(instance_key) = instance_key {
-                return Err(PathParseError::UnexpectedInstanceKey(instance_key));
-            }
-            if let Some(component_name) = component_name {
-                return Err(PathParseError::UnexpectedComponentName(component_name));
-            }
-            Ok(entity_path)
-        } else {
-            let mut parts = parse_entity_path_forgiving(s)?;
+        if let Some(instance_key) = instance_key {
+            return Err(PathParseError::UnexpectedInstanceKey(instance_key));
+        }
+        if let Some(component_name) = component_name {
+            return Err(PathParseError::UnexpectedComponentName(component_name));
+        }
+        Ok(entity_path)
+    }
 
-            for part in &mut parts {
-                if let EntityPathPart::Name(name) = part {
-                    if validate_name(name).is_err() {
-                        // Quote this, e.g. `foo/invalid.name` -> `foo/"invalid.name"`
-                        *part = EntityPathPart::Index(Index::String(name.to_string()));
-                    }
+    /// Parses an entity path, handling any malformed input with a logged warning.
+    ///
+    /// For a strict parses, use [`Self::parse_strict`] instead.
+    pub fn parse_forgiving(s: &str) -> Self {
+        let mut parts = parse_entity_path_forgiving(s);
+
+        for part in &mut parts {
+            if let EntityPathPart::Name(name) = part {
+                if validate_name(name).is_err() {
+                    // Quote this, e.g. `foo/invalid.name` -> `foo/"invalid.name"`
+                    *part = EntityPathPart::Index(Index::String(name.to_string()));
                 }
             }
+        }
 
-            let path = EntityPath::from(parts);
-
-            if path.to_string() != s {
-                re_log::warn_once!("Found an invalid entity path '{s}' that was interpreted as '{path}'. Only ASCII characters, numbers, underscore, and dash are allowed in identifiers. To put wild text in an entity path, surround it with double-quotes.");
-            }
+        let path = EntityPath::from(parts);
 
-            Ok(path)
+        if path.to_string() != s {
+            re_log::warn_once!("Found an entity path '{s}' that was not in the normalized form. Please write it as '{path}' instead. Only ASCII characters, numbers, underscore, and dash are allowed in identifiers. To put wild text in an entity path, surround it with double-quotes.");
```

Comment:
```suggestion
            re_log::warn_once!("Found an entity path '{s}' that was not in the normalized form. Please write it as '{path}' instead. Only ASCII characters, numbers, underscore, and dash are allowed in identifiers. To put arbitrary text in an entity path, surround it with double-quotes.");
```

Also, this doesn't mention that quotes within need to be escaped